miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ArrayRestController.java

```diff
@@ -0,0 +1,156 @@
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.Response.Status;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Array;
+import uk.ac.bbsrc.tgac.miso.core.data.Sample;
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginatedDataSource;
+import uk.ac.bbsrc.tgac.miso.dto.ArrayDto;
+import uk.ac.bbsrc.tgac.miso.dto.ChangeLogDto;
+import uk.ac.bbsrc.tgac.miso.dto.DataTablesResponseDto;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.SampleDto;
+import uk.ac.bbsrc.tgac.miso.service.ArrayService;
+import uk.ac.bbsrc.tgac.miso.service.SampleService;
+
+@Controller
+@RequestMapping("/rest/arrays")
+public class ArrayRestController extends RestController {
+
+  @Autowired
+  private ArrayService arrayService;
+
+  @Autowired
+  private SampleService sampleService;
+
+  private final JQueryDataTableBackend<Array, ArrayDto> jQueryBackend = new JQueryDataTableBackend<Array, ArrayDto>() {
+
+    @Override
+    protected ArrayDto asDto(Array model) {
+      return Dtos.asDto(model);
+    }
+
+    @Override
+    protected PaginatedDataSource<Array> getSource() throws IOException {
+      return arrayService;
+    }
+
+  };
+
+  @RequestMapping(value = "/dt", method = RequestMethod.GET, produces = "application/json")
+  @ResponseBody
+  public DataTablesResponseDto<ArrayDto> dataTable(HttpServletRequest request, HttpServletResponse response,
+      UriComponentsBuilder uriBuilder) throws IOException {
+    return jQueryBackend.get(request, response, uriBuilder);
+  }
+
+  @RequestMapping(method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.CREATED)
+  public @ResponseBody ArrayDto save(@RequestBody ArrayDto dto) throws IOException {
+    return doSave(dto);
+  }
+
+  @RequestMapping(value = "/{arrayId}", method = RequestMethod.PUT)
+  public @ResponseBody ArrayDto update(@PathVariable(name = "arrayId", required = true) long arrayId, @RequestBody ArrayDto dto)
+      throws IOException {
+    if (dto.getId().longValue() != arrayId) {
+      throw new RestException("Array ID mismatch", Status.BAD_REQUEST);
+    }
+    Array existing = arrayService.get(arrayId);
+    if (existing == null) {
+      throw new RestException("Array not found", Status.NOT_FOUND);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Array not found" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)